roles/fio_distributed/tasks/main.yml

```diff
@@ -65,17 +65,23 @@
         - app = fio-benchmark-{{ trunc_uuid }}
     register: server_pods
 
+  - name: Check IP addresses
+    set_fact:
+      check_ip: true
+    when: "workload_args.servers|default('1')|int == (server_pods | json_query('resources[].status.podIP')|length)"
```

Comment:
This could still technically fail if server_pods was empty in which case it would dump an error.

Reply:
Lemme know if I have addressed your concerns.